checkov/yaml_doc/base_registry.py

```diff
@@ -253,3 +256,29 @@ def get_result_key(self, check: BaseCheck,
     def extract_entity_details(self, entity: dict[str, Any]) -> tuple[str, str, dict[str, Any]]:
         # not used, but is an abstractmethod
         pass
+
+    def set_lines_for_item(self, item: str) -> dict[int | str, str | int] | str:
+        if not self.definitions_raw:
+            return item
+
+        item_lines = item.rstrip().split("\n")
+        first_line, last_line = item_lines[0], item_lines[-1]
+
+        is_single_line = True if len(item_lines) == 1 else False
```

Comment:
why not a special handling for single lines, instead of trying to integrate in the code for multi lines? something like
```python
if len(item_lines) == 1:
    return {...}
```